src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -102,6 +104,12 @@ public static class ExpressionMetadataProvider
                 //    m => m.Color.Red (nested property access)
                 //    m => m.Widgets[0].Size (expression ending with property-access)
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
+                if (metadata == null)
+                {
+                    throw new InvalidOperationException(Resources.FormatModelExpression_InvalidProperty(
+                        nameof(ModelMetadata),
+                        containerType.Name + "." + propertyName));
```

Comment:
Need something better than `$"{type}.{property}"`. That isn't valid C# unless the property is `static`. It also won't mean much when the property doesn't exist at all or is a field. Suggest passing the property and type names to the resource separately.

(Personally, I'm glad we didn't have tests demonstrating an NRE! Would have been better if we'd noticed the bug of course.)